monai/networks/layers/spatial_transforms.py

```diff
@@ -116,6 +118,10 @@ def grid_pull(
     ]
     out: torch.Tensor
     out = _GridPull.apply(input, grid, interpolation, bound, extrapolate)
+    if isinstance(input, monai.data.MetaTensor):
+        meta = input.meta if input.is_batch else deepcopy(input.meta)
+        applied_ops = input.applied_operations if input.is_batch else deepcopy(input.applied_operations)
+        out = monai.data.MetaTensor(out, meta=meta, applied_operations=applied_ops)
```

Comment:
1. I think if the `out` is native PyTorch Tensor, it already has the logic to deepcopy the meta when creating:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/meta_tensor.py#L133
Or maybe I misunderstand the logic here?
2. If must deepcopy, I see 4 duplicated code logic in this file change, maybe better to add a `deepcopy` or `force_deepcopy` flag when creating a `MetaTensor` to simplify the code?

Thanks.

Reply:
sure, I didn't see there's a deepcopy in the constructor.